pyogrio/_io.pyx

```diff
@@ -1767,6 +1783,30 @@ IF CTE_GDAL_VERSION >= (3, 8, 0):
 
 IF CTE_GDAL_VERSION >= (3, 8, 0):
 
+    cdef is_geometry_field(const ArrowSchema* schema):
+        cdef const char *metadata = schema.metadata
+
+        if metadata == NULL:
+            return False
+
+        n = int.from_bytes(metadata[:4], byteorder=sys.byteorder)
+        pos = 4
+
+        for i in range(n):
+            length_key = int.from_bytes(metadata[pos:pos+4], byteorder=sys.byteorder)
+            pos += 4
+            key = metadata[pos:pos+length_key]
+            pos += length_key
+            length_value = int.from_bytes(metadata[pos:pos+4], byteorder=sys.byteorder)
+            pos += 4
+            value = metadata[pos:pos+length_value]
+            pos += length_value
+
+            if key == b"ARROW:extension:name":
+                if value == b"geoarrow.wkb" or value == b"ogc.wkb":
+                    return True
```

Comment:
It would probably also be good to check here for any geoarrow extension arrays that are _not_ `geoarrow.wkb` and error. I'm sure someone will try to pass e.g. a `geoarrow.point` array in at some point and be confused why no geometry was written.

Reply:
We currently set the CRS at the moment of creating the layer, so that is before passing any Arrow data. So I would _expect_ it cannot deal with the CRS embedded in the schema metadata at that point